examples/rust/components/blobby/src/wrapper.rs

```diff
@@ -2,111 +2,38 @@
 // something we should probably make part of wit-bindgen or something else since this will be a
 // common thing people will need in Rust
 
-use std::io;
-use std::io::Read;
-
-use crate::wasi::io::streams::StreamError;
-
-impl Read for crate::wasi::io::streams::InputStream {
-    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
-        let n = buf
-            .len()
-            .try_into()
-            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
-        match self.blocking_read(n) {
-            Ok(chunk) => {
-                let n = chunk.len();
-                if n > buf.len() {
-                    return Err(io::Error::new(
-                        io::ErrorKind::Other,
-                        "more bytes read than requested",
-                    ));
+use crate::wasi::{
+    io::streams::{InputStream, OutputStream, StreamError},
+    logging::logging::{log, Level},
+};
+
+/// Helper function to read all the data from the input stream and write it to the output stream
+/// Optionally takes a length to limit the number of bytes read from the input stream
+pub fn stream_input_to_output(
```

Comment:
how about we use https://docs.rs/wasi/latest/wasi/ and https://doc.rust-lang.org/std/io/fn.copy.html instead?